datanode/dehistory/config.go

```diff
@@ -15,8 +13,7 @@ type Config struct {
 	Enabled       encoding.Bool     `long:"enabled" description:"set to false to disable decentralized history"`
 	WipeOnStartup encoding.Bool     `long:"wipe-on-startup" description:"remove all deHistory state on startup"`
 
-	AddSnapshotsToStore  encoding.Bool     `long:"add-snapshots-to-store" description:"if true snapshot data produced by this node will be added to the decentralise history store"`
-	AddSnapshotsInterval encoding.Duration `long:"add-snapshots-interval" description:"interval between checking for and adding snapshot data to the decentralised store"`
+	Publish encoding.Bool `long:"publish" description:"if true this node will create and publish decentralized history segments"`
```

Comment:
How does this differ from:
```
Enabled       encoding.Bool     `long:"enabled" description:"set to false to disable decentralized history"`
```
?

Reply:
That prevents the node from 'reading/recovering from' dehistory and connecting into the IPFS peer network.

Reply:
but in the code it also turns off actually taking snapshots?
```
if l.conf.DeHistory.Enabled {
		l.Log.Info("Initializing Decentralized History")
		err = l.initialiseDecentralizedHistory()
		if err != nil {
			return fmt.Errorf("failed to initialise decentralized history:%w", err)
		}
	}
```
`l.snapshotService` is initialised in `initialiseDecentralizedHistory()`

Reply:
yep, publish is to control whether the node is a reader/writer of segments in essence.  Enabled just switches dehistory off altogether, so all IPFS stuff and including ability to use it to restore a node and includes switching off snapshot creation/publishing.